torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1188,14 +1188,20 @@ void InitXlaModuleBindings(py::module m) {
         runtime::GetComputationClient()->GetAllDevices();
     return all_devices;
   });
-  m.def("_xla_real_devices", [](const std::vector<std::string>& devices) {
-    std::vector<std::string> xla_devices;
-    {
-      NoGilSection nogil;
-      xla_devices = GetXlaDevices(devices);
-    }
-    return xla_devices;
-  });
+  m.def("_xla_real_devices",
+        [](const std::optional<std::vector<std::string>> devices) {
+          if (!devices) {
+            return runtime::GetComputationClient()->GetLocalDevices();
+          }
+
+          std::vector<std::string> xla_devices;
+          {
+            NoGilSection nogil;
```

Comment:
Do you know why we release the GIL in this block? Is it for tpu v2/v3 where we allow multiple threads to do some runtime job such as `GetXlaDevices(*devices)`?

Reply:
I'm not sure to be honest. Maybe `GetLocalDevices` was a blocking call in XRT?

Reply:
oh lol it is that `torch_xla`